src/transformer.js

```diff
@@ -498,12 +498,15 @@ function _renderMarkdown(htmlDoc, mediaMap) {
 
             if (original !== rendered) {
                 if (mediaMap != null) {
-                    const fragment = libxmljs.parseHtmlFragment(rendered);
-
-                    rendered = _replaceMediaSources(
-                        fragment,
-                        mediaMap
-                    ).toString(false);
+                    const fragment = libxmljs.parseHtmlFragment(
```

Comment:
Am I understanding correctly that markdown currently works fine in the absence of any media used by the form but as soon as the form uses media this mangling happens for any label/hint that uses markdown?

I haven't loaded all this context back in to my brain but the two tests above seem odd to me because they're not specifically identifying cases of media usage, it's using a proxy condition that overselects (right?). Probably no action to take now but wanted to mention that this is what jumped out at me.

Reply:
Oh nice find! It turns out mediaMap is actually never `null` (it becomes `{}` without media). I will fix that and then pass a non-empty mediaMap in the tests.

> Am I understanding correctly that markdown currently works fine in the absence of any media used by the form but as soon as the form uses media this mangling happens for any label/hint that uses markdown?

Yes, indeed.

Reply:
This bug would surface in case some application does provide `survey.media` when calling `transform`.